ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -91,6 +94,34 @@ jobs jobPool st =
       Completion completion <- JobPool.waitForJob jobPool
       return (completion st)
 
+-- | Monitor new inbound connections
+--
+newInboundConnections :: forall m peeraddr peerconn.
+                          (MonadSTM m, Ord peeraddr)
+                      => PeerSelectionActions peeraddr peerconn m
+                      -> PeerSelectionState peeraddr peerconn
+                      -> Guarded (STM m) (TimedDecision m peeraddr peerconn)
+newInboundConnections PeerSelectionActions{
+                        readNewInboundConnection
+                      }
+                      st@PeerSelectionState {
+                        knownPeers
+                      } =
+    Guarded Nothing $ do
+      (addr, ps) <- readNewInboundConnection
+      return $ \_ ->
+        let -- If peer happens to already be present in the Known Peer set
+            -- 'insert' is going to do its due diligence before adding.
+            newEntry    = Map.singleton addr (ps, DoAdvertisePeer, IsNotLedgerPeer)
```

Comment:
So we now have `PeerSharing` and `AdvertisePeer`. `PeerSharing` means if one is running the peer sharing protocol and `AdvertisePeer` is whether the peer should be shared, isn't it?  Don't we need to advertise `AdvertisePeer` through handshake too?  If a BP connects to a relay and it doesn't wishes to be advertised, how can it be achieved?

Reply:
ok, thanks.  Btw, we should document this somewhere visible to SPOs, e.g. in `cardano-node`.

Reply:
I'll make a note and as soon as I am able to write a PR for cardano node I'll include it!

Reply:
I did this on the Peer Sharing PR that made it to cardano-node. I left it in the commit message